inst/recorder/app.R

```diff
@@ -183,35 +176,54 @@ codeGenerators <- list(
 
       code
 
-    } else {
+    } else if (event$hasBinding) {
       paste0(
         "app$setInputs(",
         quoteName(event$name), " = ",
         processInputValue(event$value, event$inputType),
         args,
         ")"
       )
+
+    } else {
+      if (allowInputNoBinding) {
```

Comment:
Seems like you could get rid of a set of parentheses: `else if (allowInputNoBinding) {`